app.js

```diff
@@ -6,95 +6,107 @@ var express = require('express');
 var compression = require('compression');
 var bodyParser = require('body-parser');
 var multer = require('multer');
-var pkg_info = require('./package.json');
 var fs = BBPromise.promisifyAll(require('fs'));
 
 
 /**
- * The main application object
+ * Promise create an express app and initialize it
+ * @param options
+ * @returns {bluebird}
  */
-var app = express();
+function initApp(options) {
+    // The main application object
+    return BBPromise.resolve(express()).then(function(app){
 
-/* Basic configuration */
-
-// disable the X-Powered-By header
-app.set('x-powered-by', false);
-// disable the ETag header - users should provide them!
-app.set('etag', false);
-// enable compression
-app.use(compression({level: 3}));
-// use the JSON body parser
-app.use(bodyParser.json());
-// use the application/x-www-form-urlencoded parser
-app.use(bodyParser.urlencoded({ extended: true }));
-// use the multipart/form-data
-app.use(multer());
-// serve static files from static/
-app.use('/static', express.static(__dirname + '/static'));
-
-/*** More configuration of app comes here ***/
+        // get the options and make them available in the app
+        app.logger = options.logger;    // the logging device
+        app.metrics = options.metrics;  // the metrics
+        app.conf = options.config;      // this app's config options
+        app.info = require('./package.json'); // this app's package info
```

Comment:
It's highly probable that `package.json` is already in memory, but still, sync file reading in a promise? I think it was better before with it being loaded together with the file.